.github/workflows/ci.yml

```diff
@@ -1,14 +1,21 @@
 name: CI
 
-on:
+on:  
+  workflow_dispatch:
   pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
   push:
-    branches: main
+    branches:
+      - main
+      - nightly
```

Comment:
Note the addition of `nightly` branch here rather than running bespoke tests in the nightly build.